scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/GroupOperationsHttpResponseCodeTest.java

```diff
@@ -77,17 +77,17 @@ public void testGetGroup200() {
   }
 
   @Test
-  @DisplayName("Test Get group with illegal id and verify Http status code: 400")
-  public void testGetGroupWithIllegalId400() {
+  @DisplayName("Test Get group with illegal id and verify Http status code: 404")
+  public void testGetGroupWithIllegalIdReturns404() {
     logger.info("Fetching Group with illegal id");
     SCIMResponse<Group> scimResponse = resourceAwareGroupRequest.readSingleGroup(ILLEGAL_UUID);
 
-    assertAll("Verify GET Response", getResponseStatusAssertions(scimResponse, false, BAD_REQUEST));
+    assertAll("Verify GET Response", getResponseStatusAssertions(scimResponse, false, NOT_FOUND));
   }
```

Comment:
The current implementation does not define how the globally unique ID should look like. All modified tests that test with an "illegal" hardcoded value should be deleted.